### PR TITLE
hide the cube when hud is off

### PR DESCRIPTION
**Pull request by SwissalpS**
see https://github.com/mt-mods/xp_redo/issues/45
replaces https://github.com/mt-mods/xp_redo/pull/57 and https://github.com/mt-mods/xp_redo/pull/58

### CODE REVIEW

**Comment by SwissalpS** [comment]
have only tested in singleplayer

File: entities.lua [hunk 1]
```diff
@@ -63,6 +63,9 @@ xp_redo.update_rank_entity = function(player, rank)
 		return
 	end
 
+	local state = player:get_meta():get(xp_redo.HUD_DISPLAY_STATE_NAME)
+	if not (not state or state == "on") then return end
```
Comment (OgelGames) [inline entities.lua:67]: I think this would be simpler:
```suggestion
	if state == "off" then return end
```

**Comment by OgelGames** [comment]
Not tested but looks correct 👍

File: entities.lua (hunk 1 above)
Comment (SwissalpS) [reply entities.lua]: thanks, so true :D

**Comment by SwissalpS** [comment]
In the meantime I tested with two players and it worked. The 'cobble peasant' part of name doesn't update without re-log though.
That might be for another PR.

**Pull request merged**